tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,257 @@
+    ann_path = paths['ann_file']
+    del paths['ann_file']
+    lut = _build_lut(paths)
+    train_split, val_split, test_split = _ann_parser(ann_path)
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                "lut": lut,
+                "split": train_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                "lut": lut,
+                "split": val_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            gen_kwargs={
+                "lut": lut,
+                "split": test_split,
+            },
+        ),
+    ]
+
+
+  def _generate_examples(self, lut, split):
+    """Returns examples
+    :type lut: dict, lookup dictionary to lookup whole paths of series_folder
+    :type split: pd.dataframe, annotation information associated to images
+    :rtype: yield idx, example
+    """
+    for idx, value in enumerate(split.values):
+      file_name, bboxs, size, sp = value
+      record = {
+          "image/name": file_name,
+          "image": _lookup_image_path(lut, file_name),
+          "bbox": _format_bboxs(bboxs, size)
+      }
+      yield idx, record
+
+    
+def _build_lut(paths):
+    """Returns lookup dictionary to lookup whole paths of series_folder
+    :type paths: dict, {<zipfn>:<unzip_path>}
+    :rtype: dict, {<series_folder>:<unzip_path>/Images_png/<series_folder>}
+    """
+    lut = {}
+    for k, v in paths.items():
+        for fd in tf.io.gfile.listdir(os.path.join(v,'Images_png')):
+            lut[fd] = os.path.join(v,'Images_png', fd)
+    return lut
+        
+
+def _lookup_image_path(lut, file_name):
+    """Returns whole path of an image
+    :type lut: dict, lookup dictionary
+    :file_name: string, image name recorded in the annotation file
+    """
+    slice_fn = file_name.split('_')[-1]
+    fd = '_'.join(file_name.split('_')[:-1])
+    
+    return os.path.join(lut[fd], slice_fn)
+    
+    
+def _format_bboxs(bboxs, size):
+    """Return bbox feature
+    :type bboxs: string, "xxx,xxx,xxx,xxx"
+    :type size: string, "xxx,xxx"
+    :rtype: tfds.features.BBox
+    """
+    size = [float(x) for x in size.split(',')]
+    coords = np.clip([float(x) for x in bboxs.split(',')], 0.0, size[0])
+
+    cnt = int((len(coords))/4)
+
+    bbox_list = []
+    for i in range(cnt):
+        ymin = coords[i*4+1] / size[1]
+        xmin = coords[i*4] / size[0]
+        ymax = coords[i*4+3] / size[1]
+        xmax = coords[i*4+2] / size[0]
+        bbox_list.append(tfds.features.BBox(ymin=ymin, xmin=xmin, ymax=ymax, xmax=xmax))
+    
+    return bbox_list
+    
+    
+def _ann_parser(ann_path):
+    """parsers the annotation file and returns the splits in dataframes
+    :type ann_path: string, path of the annotation file
+    :rtype: pandas.dataframes
```

Comment:
tuple with 3 dataframes: for train, validation and test.

Reply:
> tuple with 3 dataframes: for train, validation and test.

fixed